docs/source/configuration/external_timelock_service_configs/timelock_server_config.rst

```diff
@@ -175,6 +175,50 @@ fairly.
        stream. This margin is specified as a ratio of the smallest idle timeout - hence it must be strictly between
        0 and 1 (default: ``0.03``).
 
+.. _async-lock-service:
+
+Async Lock Service
+------------------
+
+.. danger::
+
+   If disabling legacy safety checks, note that clients for each namespace **MUST** either use pre- or post-0.49.0
+   versions of AtlasDB. This also precludes rolling upgrades of clients (as there is an intermediate state where some
+   clients are using pre-0.49.0 and some using post-0.49.0). Failure to ensure this may result in
+   **SEVERE DATA CORRUPTION** as transactions which would otherwise have run into a write-write conflict might
+   successfully commit.
+
+Since AtlasDB 0.49.0, the TimeLock Server by default runs an asynchronous implementation of the lock service.
+This relies on new ``/timelock`` APIs (as opposed to the previous ``/timestamp`` and ``/lock`` APIs). Configuring
+the asynchronous lock service may be done as follows:
+
+   .. code:: yaml
+
+      asyncLock:
```

Comment:
this break is probably fine given the small blast radius, but we should remember to proactively reach out to existing deployments to make sure they have painless upgrades